source/state_representation/include/state_representation/space/cartesian/CartesianTwist.hpp

```diff
@@ -120,20 +123,6 @@ class CartesianTwist : public CartesianState {
    */
   CartesianTwist& operator=(const CartesianTwist& twist) = default;
 
-  /**
-   * @brief Overload the *= operator
-   * @param twist CartesianTwist to multiply with
-   * @return the current CartesianTwist multiplied by the CartesianTwist given in argument
-   */
-  CartesianTwist& operator*=(const CartesianTwist& twist);
```

Comment:
If we want to avoid another major version increment to control libraries 4.0, then the correct approach would be to mark these function as deprecated instead of deleting them:

```c++
[[deprecated]] CartesianTwist& operator*=(const CartesianTwist& twist);
``` 

Same goes for the base functions (e.g. `CartesianState operator*(const CartesianState& state)`), instead of deleting you would override / hide the implementation and mark it as deprecated. (hiding in this context is ok, since we don't intend the method to actually be used)

We could then remove these cases entirely the next time we decide to release a new major version.

Reply:
I get the first part but not the second one. Do I have to explicitly add the operator `CartesianState operator*(const CartesianState& state)` in the header and source (where i just call the CartesianState operator) and then mark them as deprecated in the header?

Reply:
yes exactly, for the second part you would have to re-implement those functions for this class. It may be more effort than it is worth, that's up to you and @buschbapti to decide. The first part I think still makes sense though.

Reply:
We'll see if the tests pass without, otherwise it may be necessary. I only came up with that because we had a weird behaviour in a test, where during a pose * twist multiplication, the twist was integrated to a pose and then multiplied it with pose * pose